### ISSUE TITLE
Runtime compilation?

### ISSUE THREAD

**Issue by Infinitybeond1** [enthus1ast/nimja#79]
Is it possible to compile these templates at runtime? with a function or something?

**Comment by enthus1ast** [comment]
Currently there is no real good way, the only semi good option is hot code reloading, so that your application checks if the templates has changed, then calls the nim compiler to recompile the templates to a dll, unloads and reloads this dll. 
See:
https://github.com/enthus1ast/nimja/tree/master/examples/hcr
and
https://github.com/enthus1ast/nimja#automatic-recompilation--hot-code-reloading-hcr

The benefit of this is, that they are as fast as normal templates, the downside is that you must restructure your application a little.

Real Runtime evaluation however is still a planned feature, and there is code for this already in the "dynamicAgain" branch (and some more i have not published yet).
The way i want to go is to include the compiler as a library, let nimja generate PNodes (instead of NimNodes) and execute these directly in the nimvm.  The challenge here is, that the compiler api is quite undocumented and complicated to use (for me at least); and the other challenge is to convert parameters from native nim types, to PNodes that can run directly on the vm.

So in short, i would love to have this as well, but its not yet done.
